yktracec/src/jitmodbuilder.cc

```diff
@@ -1806,10 +1806,12 @@ class JITModBuilder {
               VMap[CI] = ConstantPointerNull::get(
                   Type::getInt8PtrTy(JITMod->getContext()));
 
-              // The next two trace blocks are the (unmappable) control point
-              // followed by the return to this block which only has a branch
-              // instruction left which we don't care about. So we can just skip
-              // over these and continue with the block after.
+              // The next two trace blocks may be skpped as they only contain
+              // the (unmappable) control point followed by the return to the
+              // current block which only has a branch instruction left which we
+              // don't care about. Skipping the latter also means we don't
+              // create additional guard-failure blocks for branches that can
+              // never fail, reducing the trace even more.
```

Comment:
Presumably the only branch that can never fail is the very final branch in a trace?

Reply:
I see!